ruma-federation-api/src/membership/create_invite/v1.rs

```diff
@@ -0,0 +1,157 @@
+        name: "create_invite",
+        path: "/_matrix/federation/v1/invite/:room_id/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The room ID that the user is being invited to.
+        #[serde(rename = "roomId")]
+        #[ruma_api(path)]
+        room_id: RoomId,
+
+        /// The event ID for the invite event, generated by the inviting server.
+        #[serde(rename = "eventId")]
+        #[ruma_api(path)]
+        event_id: EventId,
+
+        /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+        sender: UserId,
+
+        /// The name of the inviting homeserver.
+        origin: Box<ServerName>,
+
+        /// A timestamp added by the inviting homeserver.
+        origin_server_ts: UInt,
+
+        /// The value `m.room.member`.
+        #[serde(rename = "type")]
+        kind: EventType,
+
+        /// The user ID of the invited member.
+        state_key: UserId,
+
+        /// The content of the event.
+        content: MemberEventContent,
+
+        /// Information included alongside the event that is not signed.
+        unsigned: UnsignedEventContent,
+    }
+
+    response: {
+        /// The response invite event
+        #[ruma_api(body)]
+        #[serde(with = "crate::serde::invite_response")]
+        invite_response: InviteResponse,
+    }
+}
+
+/// Information included alongside an event that is not signed.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UnsignedEventContent {
+    /// An optional list of simplified events to help the receiver of the invite identify the room.
+    /// The recommended events to include are the join rules, canonical alias, avatar, and name of
+    /// the room.
+    invite_room_state: Vec<StrippedState>,
+}
+
+/// A simplified event that helps the server identify a room.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct StrippedState {
+    /// The `content` for the event.
+    content: Value,
+
+    /// The `state_key` for the event.
+    state_key: String,
+
+    /// The `type` for the event.
+    #[serde(rename = "type")]
+    kind: EventType,
+
+    /// The `sender` for the event.
+    sender: UserId,
+}
+
+/// The invite event sent as a response.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct InviteResponse {
+    /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+    sender: UserId,
+
+    /// The name of the inviting homeserver.
+    origin: Box<ServerName>,
+
+    /// A timestamp added by the inviting homeserver.
+    origin_server_ts: UInt,
+
+    /// The event type (should always be `m.room.member`).
+    kind: EventType,
+
+    /// The user ID of the invited member.
+    state_key: UserId,
+
+    /// The content of the event. Must include a `membership` of invite.
+    content: MemberEventContent,
+}
+
+impl Request {
+    /// Creates a new `Request` with the given parameters.
+    pub fn new(
```

Comment:
For a type with this many fields, let's create an [init struct](https://github.com/ruma/ruma/blob/1fbf05d71ffa94f0571ca9850aa5c112ddd85dff/ruma-common/src/push.rs#L77).